pulsar-client/src/main/java/org/apache/pulsar/client/api/PulsarClient.java

```diff
@@ -43,6 +46,17 @@ public static ClientBuilder builder() {
         return new ClientBuilderImpl();
     }
 
+    /**
+     * Create a pulsar client builder instance with a provided configuration map.
+     *
+     * @param config configuration map.
+     * @return client builder
+     * @since 2.1.0
+     */
+    static ClientBuilder builder(Map<String, Object> config) {
```

Comment:
Why not use the same approach used for `ProducerBuilder` and `ConsumerBuilder` and have `ClientBuilder.loadConf()`?

Reply:
That would be quite a few overloaded builders for consumer. Are you okay with that?

- builder()
- builder(schema)
- builder(config)
- builder(schema, config)

if you are okay with that, I can go down into that route. then your subsequent comment will be addressed as well.

Reply:
I mean the reverse :) 

```java
PulsarClient client = PulsarClient.builder().loadConf(..)
client.newProducer().loadConf(xxx);
```